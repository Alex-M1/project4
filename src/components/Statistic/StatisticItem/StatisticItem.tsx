/* eslint-disable no-plusplus */
import { LOCAL_STORAGE } from 'constants/constants';
import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import { useTranslation } from 'react-i18next';

interface IProps {
  creatorLogin: string, 
  gameType: string, 
  guestLogin: string, 
  winnerLogin: string,
}

const StatisticItem: React.FC <IProps> = ({ creatorLogin, gameType, guestLogin, winnerLogin }) => {
  const { t } = useTranslation();
  const userLogin = localStorage.getItem(LOCAL_STORAGE.login);
  return (
      <div key={uuidv4()}>
        <span>{userLogin}</span>
        <span>{t(gameType)}</span>
        <span>{userLogin === creatorLogin ? guestLogin : creatorLogin}</span>
        <span>{winnerLogin}</span>
      </div>
  );
};

export default StatisticItem;
