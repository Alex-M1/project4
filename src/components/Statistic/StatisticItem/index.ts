import { connect } from 'react-redux';
import { STATISTIC_DATA } from '../../../../__mocks__/statisticByUUID';
import StatisticItem from './StatisticItem';

const mapStateToProps = () => ({
  statistic: STATISTIC_DATA,
});

export default connect(mapStateToProps)(StatisticItem);
