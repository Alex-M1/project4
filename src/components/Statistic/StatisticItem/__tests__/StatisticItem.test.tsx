import React from 'react';
import { mount, shallow } from 'enzyme';
import StatisticItem from '../StatisticItem';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
        t: jest.fn(),
    })),
}),
);

describe('StatisticItem', () => {
    let props;
    beforeEach(() => {
        props = {
          creatorLogin, 
          gameType, 
          guestLogin, 
          winnerLogin
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<StatisticItem {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StInputDiv', () => {
        const component = mount(<StatisticItem {...props} />);
        expect(component.find('styled__StInputDiv')).toHaveLength(1);
    });
});
