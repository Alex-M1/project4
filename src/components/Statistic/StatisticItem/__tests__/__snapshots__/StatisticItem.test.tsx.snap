// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticItem Should match snapshot 1`] = `"<div><span></span><span></span><span>string</span><span>string</span></div>"`;
